src/Raven.Server/Documents/Sharding/Handlers/Processors/Replication/ShardedReplicationHandlerProcessorForGetConflicts.cs

```diff
@@ -59,18 +58,27 @@ public ShardedGetReplicationConflictsOperation(ShardedDatabaseRequestHandler han
 
             public GetConflictsPreviewResult Combine(Memory<GetConflictsPreviewResult> results)
             {
-                var span = results.Span;
-                var final = new GetConflictsPreviewResult();
-
-                final.Results = _handler.DatabaseContext.Streaming.PagedShardedItem(
-                    results,
-                    selector: r => r.Results,
-                    comparer: ConflictsLastModifiedComparer.Instance,
-                    _token).ToArray();
+                var totalResults = 0L;
+                foreach (var conflictResult in results.Span)
+                    totalResults += conflictResult.TotalResults;
+                
+                var final = new GetConflictsPreviewResult
+                {
+                    Results = new List<GetConflictsPreviewResult.ConflictPreview>(),
+                    TotalResults = totalResults
+                };
 
-                foreach (var s in span)
+                var pageSize = _token.PageSize;
+                foreach (var res in _handler.DatabaseContext.Streaming.CombinedResults(results, r => r.Results, ConflictsLastModifiedComparer.Instance))
                 {
-                    final.TotalResults += s.TotalResults;
+                    final.Results.Add(res.Item);
+                    pageSize--;
+
+                    if (pageSize <= 0)
+                        break;
+
+                    var shard = res.Shard;
+                    _token.Pages[shard].Start += (int)res.Item.ScannedResults;
```

Comment:
is `+=` correct? if you collect 2 results here next time you will skip here more than intended, no?.